### ISSUE TITLE
Add ENUM type to H2DataType

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9254]
Support for H2 `ENUM` type was implemented with #5874. But it looks like `H2DataType` wasn't modified to include an `ENUM` type.

**Comment by lukaseder** [comment]
We don't have such a type in `SQLDataType`. I'm not sure what value it would add, currently?

**Comment by knutwannheden** [comment]
> We don't have such a type in `SQLDataType`. I'm not sure what value it would add, currently?

We do have the `ENUM` type for other dialects like MySQL. What got my attention was a lot of logged `Type ENUM is not supported in dialect H2` messages when running our integration tests against H2.

**Comment by lukaseder** [comment]
> We do have the ENUM type for other dialects like MySQL

You're right - huh. Why do we have that...? I don't see its value. Well, I'll look into this next week. I don't think we should have this type for MySQL either.
